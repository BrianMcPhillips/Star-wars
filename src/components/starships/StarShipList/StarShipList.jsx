import React from 'react';
import PropTypes from 'prop-types';
import StarShip from '../StarShip/StarShip';

const StarShipList = ({ data }) => {
  const shipStuff = data.map((ship, i) => 
    <li key={ship + i}>
      <StarShip {...ship}/>
    </li>
  );

  return (
    <ul>
      {shipStuff}
    </ul>
  );
};

StarShipList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    model: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default StarShipList;
