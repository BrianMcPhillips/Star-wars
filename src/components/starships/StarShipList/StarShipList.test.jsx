import React from 'react';
import { render, cleanup } from '@testing-library/react';
import StarShipList from './StarShipList';

describe('StarShipList component', () => {
  afterEach(() => cleanup());
  it('renders StarShipList', () => {
    const { asFragment } = render(
      <StarShipList data= {[
        {
          name: 'Star Destroyer',
          model: 'Imperial I-class Star Destroyer',
          url: 'https://swapi.dev/api/starships/3/'
        },
        {
          name: 'Sentinel-class landing craft',
          model: 'Sentinel-class landing craft',
          url: 'https://swapi.dev/api/starships/5/'
        }
      ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
