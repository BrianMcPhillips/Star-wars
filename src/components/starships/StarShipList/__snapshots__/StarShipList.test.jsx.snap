// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarShipList component renders StarShipList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h1>
        Star Destroyer
      </h1>
      <h3>
        Imperial I-class Star Destroyer
      </h3>
    </li>
    <li>
      <h1>
        Sentinel-class landing craft
      </h1>
      <h3>
        Sentinel-class landing craft
      </h3>
    </li>
  </ul>
</DocumentFragment>
`;
